linera-faucet/server/src/lib.rs

```diff
@@ -203,27 +150,36 @@ where
 }
 
 #[async_graphql::Object(cache_control(no_cache))]
-impl MutationRoot {
+impl<S> MutationRoot<S>
+where
+    S: Storage + Send + Sync + 'static,
+{
     /// Creates a new chain with the given authentication key, and transfers tokens to it.
     async fn claim(&self, owner: AccountOwner) -> Result<ChainDescription, Error> {
         self.do_claim(owner).await
     }
 }
 
-impl MutationRoot {
+impl<S> MutationRoot<S>
+where
+    S: Storage + Send + Sync + 'static,
+{
     async fn do_claim(&self, owner: AccountOwner) -> Result<ChainDescription, Error> {
-        // Check if this owner already has a chain
+        // Check if this owner already has a chain.
+        if let Some(existing_chain_id) = self
+            .faucet_storage
+            .get_chain_id(&owner)
+            .await
+            .map_err(|e| Error::new(e.to_string()))?
         {
-            let storage = self.faucet_storage.lock().await;
-            if let Some(existing_description) = storage.get_chain(&owner) {
-                return Ok(existing_description.clone());
-            }
+            // Retrieve the chain description from local storage
+            return get_chain_description_from_storage(&self.storage, existing_chain_id).await;
```

Comment:
I'd put chain descriptions into sqlite if we're at it. It'd be even faster.

Reply:
Is it plausible that a chain description lookup in the local database is a significant fraction of that? The most common case will be that someone requests a _new_ chain anyway, and then there is no lookup?